src/NuGet.Core/NuGet.LibraryModel/PublicAPI.Shipped.txt

```diff
@@ -1,76 +1,76 @@
 #nullable enable
 NuGet.LibraryModel.CentralPackageVersion
-~NuGet.LibraryModel.CentralPackageVersion.CentralPackageVersion(string name, NuGet.Versioning.VersionRange versionRange) -> void
-~NuGet.LibraryModel.CentralPackageVersion.Equals(NuGet.LibraryModel.CentralPackageVersion other) -> bool
-~NuGet.LibraryModel.CentralPackageVersion.Name.get -> string
-~NuGet.LibraryModel.CentralPackageVersion.VersionRange.get -> NuGet.Versioning.VersionRange
+NuGet.LibraryModel.CentralPackageVersion.CentralPackageVersion(string! name, NuGet.Versioning.VersionRange! versionRange) -> void
```

Comment:
Just curious, what's our policy on this?  I wonder if these should be moved to the unshipped category instead since they're not technically shipped?  I guess this was automatically done by the analyzer?  What's your opinion?

Reply:
These API have already shipped. This `CentralPackageVersion` constructor already existed. The method signature didn't change, except for being annotated. Since that specific method already also would throw ArgumentNullExceptions when a null is passed in, there's little chance that anyone will experience a breaking change when they upgrade to the version of this package with annotations. Other APIs which don't throw ArgumentNullExceptions have higher change to cause compile time errors for people upgrading, but only if their own projects enable nullable checks. If their projects don't enable nullable, then our package being annotated will have no impact on them.  Adding the `required` modifier is the most significant breaking change, but those properties already existed, and the PublicApiAnalizer didn't want to decorate the method signature in these PublicAPI files.

And null annotations are a compile-time only concept, there's no ABI changes, so binaries that were compiled against older versions of our package will still reference these methods when our dll is replaced with the newer version.

IMO, the goal of the Unshipped file is to signal that it's no risk of breaking anyone if APIs listed in that file are changed before the next release. However, as explained, that's not true for these APIs. The annotations are more "superficial", rather than "structural".